ee/tasks/test/test_status_report.py

```diff
@@ -0,0 +1,89 @@
+from datetime import datetime
+from uuid import uuid4
+
+from freezegun.api import freeze_time
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.models.event import create_event
+from ee.clickhouse.models.person import create_person_distinct_id
+from posthog.models.person import Person
+from posthog.models.team import Team
+from posthog.models.utils import UUIDT
+from posthog.tasks.status_report import status_report
+from posthog.tasks.test.test_status_report import factory_status_report
+
+
+def _create_event(distinct_id: str, event: str, lib: str, created_at: datetime, team: Team):
+    create_event(
+        event_uuid=uuid4(),
+        event=event,
+        distinct_id=distinct_id,
+        timestamp=created_at,
+        team=team,
+        properties={"$lib": lib},
+    )
+
+
+def _create_person(distinct_id: str, team: Team) -> Person:
+    person = Person.objects.create(team=team, distinct_ids=[distinct_id])
+    return Person(id=person.uuid)
+
+
+class TestStatusReport(factory_status_report(_create_event, _create_person)):  # type: ignore
+    # CH only
+    def test_status_report_duplicate_distinct_ids(self) -> None:
+        create_person_distinct_id(self.team.id, "duplicate_id1", str(UUIDT()))
+        create_person_distinct_id(self.team.id, "duplicate_id1", str(UUIDT()))
+        create_person_distinct_id(self.team.id, "duplicate_id2", str(UUIDT()))
+        create_person_distinct_id(self.team.id, "duplicate_id2", str(UUIDT()))
+        create_person_distinct_id(self.team.id, "duplicate_id2", str(UUIDT()))
+
+        for index in range(0, 2):
+            sync_execute(
+                "INSERT INTO person_distinct_id SELECT %(distinct_id)s, %(person_id)s, %(team_id)s, 1, %(timestamp)s, 0 VALUES",
```

Comment:
nit: I sort of get why but it's a bit weird that we create distinct ids in different ways

Reply:
yeah the other method didn't allow me to set timestamps and I didn't want to edit the production code just for the test